receiver/postgresqlreceiver/scraper_test.go

```diff
@@ -36,196 +36,239 @@ func TestScraper(t *testing.T) {
 	factory := new(mockClientFactory)
 	factory.initMocks([]string{"otel"})
 
-	cfg := createDefaultConfig().(*Config)
-	cfg.Databases = []string{"otel"}
-	cfg.Metrics.PostgresqlDeadlocks.Enabled = true
-	cfg.Metrics.PostgresqlTempFiles.Enabled = true
-	cfg.Metrics.PostgresqlSequentialScans.Enabled = true
-	cfg.Metrics.PostgresqlDatabaseLocks.Enabled = true
-	scraper := newPostgreSQLScraper(receivertest.NewNopCreateSettings(), cfg, factory)
-
-	actualMetrics, err := scraper.scrape(context.Background())
-	require.NoError(t, err)
-
-	expectedFile := filepath.Join("testdata", "scraper", "otel", "expected.yaml")
-	expectedMetrics, err := golden.ReadMetrics(expectedFile)
-	require.NoError(t, err)
+	runTest := func(separateSchemaAttr bool, file string) {
+		cfg := createDefaultConfig().(*Config)
+		cfg.Databases = []string{"otel"}
+		cfg.Metrics.PostgresqlDeadlocks.Enabled = true
+		cfg.Metrics.PostgresqlTempFiles.Enabled = true
+		cfg.Metrics.PostgresqlSequentialScans.Enabled = true
+		cfg.Metrics.PostgresqlDatabaseLocks.Enabled = true
+
+		scraper := newPostgreSQLScraper(receivertest.NewNopCreateSettings(), cfg, factory)
+		scraper.separateSchemaAttr = separateSchemaAttr
```

Comment:
Rather than setting this manually, we can test that instantiating the scraper actually respects the feature gate automatically. It looks like it already does so we should be able to just [set the gate](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1b6a4517b5323e7cbd9ad97e1c7e5a8e2a9054e4/internal/common/testutil/testutil.go#L129) prior to instantiation. We should do this in all cases below as well.

Reply:
I updated both tests to use testutil instead of manually setting the feature gate, both false and true so it's tested being set both ways as mentioned